### PR TITLE
Azure DDoS protection #107

### PR DESCRIPTION
**Pull request by fbouteruche**
Pull request for the issue #107 

Add ddosSwitch parameter to enable the deployment of a Azure DDoS protection plan in azuredepoy.json

Add a ddosWitch member in the moodleCommon object.
Add a nested template network-vnet-ddos.json that manages the creation
of ddos protection plan and the update of the vnet.
Call network-vnet-ddos.json from the network.json template if ddosSwith is true

In the .gitignore file, exlude .vs to ignore Visual Studio stuff.

### CODE REVIEW

File: nested/network-vnet-ddos.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            "name": "[parameters('moodleCommon').ddosPlanName]",
+            "type": "Microsoft.Network/ddosProtectionPlans"
+        },
+        {
+            "apiVersion": "2018-02-01",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/ddosProtectionPlans', parameters('moodleCommon').ddosPlanName)]"
+            ],
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('vnetName')]",
```
Comment (hosungsmsft) [inline nested/network-vnet-ddos.json]: This is great. So you did confirm that this works regardless of whether this vnet already exists or is to be created, right?

File: nested/network-vnet-ddos.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    "addressPrefixes": [
+                        "[parameters('vNetAddressSpace')]"
+                    ]
+                },
+                "ddosProtectionPlan": {
+                    "id": "[resourceId('Microsoft.Network/ddosProtectionPlans', parameters('moodleCommon').ddosPlanName)]"
+                },
+                "enableDdosProtection": "[parameters('moodleCommon').ddosSwitch]"
+            },
+            "type": "Microsoft.Network/virtualNetworks"
```
Comment (hosungsmsft) [inline nested/network-vnet-ddos.json]: This is really minor, but ARM team prefers properties like this (`type`) to be at the top, so please move this to before `properties` property.

File: nested/network-vnet-ddos.json (hunk 1 above)
Comment (fbouteruche) [reply nested/network-vnet-ddos.json]: Yes, I've ran 4 tests 
1. existing vNet and ddosSwitch equal to true
2. existing vNet and ddosSwitch equal to false
3. no existing vNet and ddosSwitch equal to true
4. no existing vNet and ddosSwitch equal to false
Each one of these 4 deployment succeed and I can browse the homepage.

Comment (hosungsmsft) [reply nested/network-vnet-ddos.json]: 💯

Comment (fbouteruche) [reply nested/network-vnet-ddos.json]: I have added documentation in the Parameters.md file and the Manage.md file.

Comment (hosungsmsft) [reply nested/network-vnet-ddos.json]: Thank you so much for your quick follow-up. This is super! I'm merging this PR right after. Thanks again!

Comment (fbouteruche) [reply nested/network-vnet-ddos.json]: You're welcome. Thanks for the great work you're doing !

Comment (joseluisrc) [reply nested/network-vnet-ddos.json:39]: Is DDoS protection included with all deployments automatically?

**Pull request merged**